tests/test_models.py

```diff
@@ -45,3 +51,227 @@ def test_search_keywords_should_not_accept_large_input(self):
+        entity1 = models.EntityModel(**response_json)
+        # local_config_response
+        entity2 = models.EntityModel.get_entity_from_yaml(local_config)
+
+        return entity1, entity2
+
+    @pytest.fixture
+    def mock_boto3(self):
+        with patch("awsmp.models.boto3") as mock_boto3:
+            mock_boto3.client.return_value.describe_regions.return_value = {
+                "Regions": [
+                    {"Endpoint": "ec2.us-east-1.amazonaws.com", "RegionName": "us-east-1", "OptInStatus": "opted-in"},
+                    {"Endpoint": "ec2.us-east-2.amazonaws.com", "RegionName": "us-east-2", "OptInStatus": "opted-in"},
+                ]
+            }
+            yield mock_boto3
+
+    def test_valid_response(self):
+        with open("./tests/test_config.json", "r") as f:
+            response_json = json.load(f)
+
+        entity_model = models.EntityModel(**response_json)
+        assert entity_model.Description.ProductTitle == "test"
+
+    def test_yaml_to_entity(self, mock_boto3):
+        with open("./tests/test_config.yaml", "r") as f:
+            local_config = yaml.safe_load(f)
+
+        entity_model = models.EntityModel.get_entity_from_yaml(local_config)
+        assert entity_model.Description.ProductTitle == "test"
+
+    def test_non_valid_response(self):
+        non_valid_response: dict[str, Any] = {
+            "Description": {
+                "ProductTitle": "test",
+                "ProductCode": "prod-test",
+            },
+            "PromotionalResources": {
+                "LogoUrl": "https://test-logourl",
+            },
+            "RegionAvailability": {"FutureRegionSupport": "All", "Restrict": [], "Regions": ["us-east-1", "us-east-2"]},
+            "SupportInformation": {"Description": "test_support_description", "Resources": ["test_support_resource"]},
+            "Versions": {"version1": ""},
+        }
+        with pytest.raises(ValidationError):
+            models.EntityModel(**non_valid_response)
+
+    @pytest.mark.parametrize(
+        "name, value1, value2, expected",
+        [
+            ("test1", "", "test", models.DiffAddedModel(name="test1", value="test")),
+            ("test2", "test", "", models.DiffRemovedModel(name="test2", value="test")),
+            (
+                "test3",
+                "test",
+                "testtest",
+                models.DiffChangedModel(name="test3", old_value="test", new_value="testtest"),
+            ),
+        ],
+    )
+    def test_get_diff_model_type(self, name, value1, value2, expected):
+        res = models.EntityModel.get_diff_model_type(name, value1, value2)
+        assert res == expected
+
+    @pytest.mark.parametrize(
+        "custom_config, expected_diff",
+        [
+            (
+                {},
+                models.DiffModel(added=[], removed=[], changed=[]),
+            ),
+        ],
+    )
+    def test_get_changed_no_diff(self, mock_boto3, get_entity, custom_config, expected_diff):
```

Comment:
We only have one case in the parametrization. If we don't need any other cases it would be better to drop the parametrize altogether